corehq/apps/translations/tests/test_updating_app_from_trans_dict.py

```diff
@@ -0,0 +1,83 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+from io import BytesIO
+
+import six
+from django.test import SimpleTestCase
+
+from corehq.apps.app_manager.models import Application
+from corehq.apps.app_manager.ui_translations import (
+    get_default_translations_for_download,
+    process_ui_translation_upload,
+)
+from corehq.apps.translations.utils import update_app_translations_from_trans_dict
+from couchexport.export import export_raw
+
+EXPECTED_TRANSLATIONS = {
+    'en': {
+        'key.manage.title': 'Taylor',
+        'bulk.send.dialog.progress': 'Kanye',
+        'specific.item.to.translate': 'Miley',
+    },
+    'fra': {
+        'key.manage.title': 'Swift',
+        'bulk.send.dialog.progress': 'West',
+        'specific.item.to.translate': 'Cyrus',
+    }
+}
+
+
+class TestBulkUiTranslation(SimpleTestCase):
+
+    def setUp(self):
+        super(TestBulkUiTranslation, self).setUp()
+        self.app = Application.new_app("test-domain", "Test App")
+        self.app.langs = ["en", "fra"]
+        self.app.translations = EXPECTED_TRANSLATIONS
+
+    def _build_translation_download_file(self, headers, data=None):
+        if data is None:
+            data = []
+            translations = get_default_translations_for_download(self.app, 'latest')
+            for translation_key, translation_value in six.iteritems(translations):
+                data.append((translation_key, translation_value))
+
+        data = (('translations', tuple(data)),)
+        temp = BytesIO()
+        export_raw(headers, data, temp)
+        temp.seek(0)            # .read() is used somewhere so this needs to be at the begininng
+        return temp
+
+    def test_not_upload_all_properties(self):
+        headers = (('translations', ('property', 'en', 'fra')),)
+        data = (('key.manage.title', 'Taylor', 'Swift'),)
+        f = self._build_translation_download_file(headers, data)
+
+        translations, error_properties, warnings = process_ui_translation_upload(self.app, f)
+        update_app_translations_from_trans_dict(self.app, translations)
+
+        self.assertDictEqual(self.app.translations, EXPECTED_TRANSLATIONS)
```

Comment:
This test currently fails, and I think its the strategy that they want to support. Can you verify that?

Reply:
Just to be sure, i don't see this as specific to REACH/icds. I think the expected behaviour is that the translations already present in the app should stay and the ones in the excel should either get added or updated.

So here in the setup we should start with,
```
CURRENT_TRANSLATIONS = {
    'en': {
        'key.manage.title': 'Taylor',
        'should.be.removed': 'Unwanted Translation',
    }
}
````
and then the update with `EXPECTED_TRANSLATIONS` in the sheet does the update correctly.
We should confirm this behaviour once with product.